gradio/processing_utils.py

```diff
@@ -323,10 +310,55 @@ def file_to_json(file_path):
     return json.load(open(file_path))
 
 
+class TempFileManager:
```

Comment:
Working now

Reply:
I think the behavior of this demo is different when compared to main.

When you click submit again, the same files as the previous generation show up at first but then they get replaced with the new generation.

![gallery_tmp_files](https://user-images.githubusercontent.com/41651716/207665252-f774f086-bc96-4bfd-85e9-30016f2f6e3e.gif)


On main, you get new images immediately

![gallery_tmp_files_main](https://user-images.githubusercontent.com/41651716/207665359-b5de05e3-0247-4cc7-8f25-11af78e1deb7.gif)

Reply:
Nice catch!!

Reply:
Ok so the reason this is happening is because we no longer download images from a URL if we don't need to. So previously, we would download the image from the unsplash to a temporary file, and then serve the image from the temporary file in the Gallery. This is not good because (1) it unnecessarily downloads files from a URL (2) it slows down the inference process. Now, for outputs, we simply pass through the URL and the Gallery component displays the URLs. However, this means that it spends a little bit of time fetching the image from the URL. Coupled with lazy loading of images and the fact that the Gallery doesn't seem to "destroy" its children images, we get the behavior shown here. 

I think this is still preferable to what we had before. Perhaps the Gallery component should "destroy" the images instead of just replacing the URLs, but that requires some frontend shenanigans. I think this is a relatively rare case so we don't need to worry about it unless we hear otherwise from users

Reply:
Ok I think that's ok since it only happens with urls and urls + gallery are pretty rare probably! Thanks for looking into it!